### PR TITLE
Tests - Use base image for frontend tests

### PR DESCRIPTION
**Pull request by Ark-kun**
This change moves the unchanging parts of the image to a base image so that they're not rebuilt every time. This improves performance and reliability.

The `Dockerfile` is now just
```
FROM gcr.io/ml-pipeline-test/selenium-standalone-chrome-gcloud-nodejs
COPY --chown=seluser . /src
WORKDIR /src
ENTRYPOINT [ "./run_test.sh" ]
```

This change is [Reviewable](https://reviewable.io/reviews/kubeflow/pipelines/190)

### CODE REVIEW

**Comment by Ark-kun** [comment]
> How much time does this actually save? Looks to me like too many lines of code if they just save us a few seconds of test time.

1. This WIP, so there are some extra lines of code that should be gone before review.
2. This should save 90% of image building time
3. This should reduce test flakiness by not downloading from Internet during build.
4. This would allow us to use local image building tools like Bazel. These tools support adding files and packages to the image, but not RUNning programs.

**Comment by yebrahim** [comment]
Doing `docker pull node:9.4.0-alpine` works for me; the image is still out there. We need to update it to use the lts version, but it's not breaking tests yet as far as I can see.

**Comment by Ark-kun** [comment]
This now works after the recent test system revamp. (The problem was that `gcloud auth configure-docker` was only called before `docker push` instead of before `docker build`)

**Comment by vicaire** [comment]
@Ark-kun, do you still need this one? Should we close?

**Comment by Ark-kun** [comment]
It's finally working as expected =)

Lessons learned:

* `COPY` always copies as `root` (not the active `USER`) unless the user the explocit `--chown=user` flag is added
* `$HOME` can point to some user's home dir even when the root is the active user. So when root user writes to `$HOME` the files can be created in other user's home directory and won't be accessible by that user
* `gcloud` write to home directory, so do not run it as root if you want to run `gcloud` from other users too.

**Comment by Ark-kun** [comment]
@yebrahim @rileyjbauer 
Should we merge this PR or close it?

File: test/frontend-integration-test/Dockerfile [hunk 1]
```diff
@@ -1,18 +1,8 @@
-FROM selenium/standalone-chrome
+FROM gcr.io/ml-pipeline-test/selenium-standalone-chrome-gcloud-nodejs:v20190507-0.1.19-19-g0b8b4dd7-dirty-20a6f9
```
Comment (rileyjbauer) [inline test/frontend-integration-test/Dockerfile]: Is this image built manually? How often do we need to rebuild this?

Comment (Ark-kun) [reply test/frontend-integration-test/Dockerfile]: > Is this image built manually?

Yes. It's build using the provided Makefile which just builds the Dockerfile and pushes the image.

>How often do we need to rebuild this?

Only if you want to upgrade Selenium, Chrome, Node etc.

File: test/frontend-integration-test/selenium-standalone-chrome-gcloud-nodejs.Docker/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+        gnupg \
+    && easy_install -U pip && \
+    pip install -U crcmod && \
+    export CLOUD_SDK_REPO="cloud-sdk-$(lsb_release -c -s)" && \
+    echo "deb https://packages.cloud.google.com/apt $CLOUD_SDK_REPO main" > /etc/apt/sources.list.d/google-cloud-sdk.list && \
+    curl https://packages.cloud.google.com/apt/doc/apt-key.gpg | apt-key add - && \
+    apt-get update && apt-get install -y google-cloud-sdk=${CLOUD_SDK_VERSION}-0 $INSTALL_COMPONENTS
+
+
+RUN apt-get update -y && \
```
Comment (yebrahim) [inline test/frontend-integration-test/selenium-standalone-chrome-gcloud-nodejs.Docker/Dockerfile]: nit: Why update again?

File: test/frontend-integration-test/selenium-standalone-chrome-gcloud-nodejs.Docker/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    curl --silent --show-error --location https://deb.nodesource.com/setup_8.x | bash - && \
+    apt-get install -y nodejs && \
+    rm -rf /var/lib/apt/lists/* /tmp/* /usr/share/locale/* /usr/share/i18n/locales/*
```
Comment (yebrahim) [inline test/frontend-integration-test/selenium-standalone-chrome-gcloud-nodejs.Docker/Dockerfile]: This isn't going to change the image size at this point. It needs to be merged with the previous layer.

File: test/frontend-integration-test/selenium-standalone-chrome-gcloud-nodejs.Docker/Dockerfile (hunk 2 above)
Comment (Ark-kun) [reply test/frontend-integration-test/selenium-standalone-chrome-gcloud-nodejs.Docker/Dockerfile]: Removed.

File: test/frontend-integration-test/selenium-standalone-chrome-gcloud-nodejs.Docker/Dockerfile (hunk 3 above)
Comment (Ark-kun) [reply test/frontend-integration-test/selenium-standalone-chrome-gcloud-nodejs.Docker/Dockerfile]: That's true >_<

**Comment by Bobgy** [comment]
@Ark-kun what's your plan about this PR?

**Comment by Ark-kun** [comment]
> @Ark-kun what's your plan about this PR?

As the Frontend area lead, you can decide whether to close it or merge.

This change was designed to improve the build speed and stability. The build speed is now pretty good even without this PR.

**Comment by Bobgy** [comment]
I think this makes code base cleaner in separating env setup with actual test steps. Also great work automating image building. Let's merge this as long as tests are still passing.

**Pull request merged**